experiments/duke/graphql_endpoint/elastic/searcher.go

```diff
@@ -80,15 +130,29 @@ func FindPublications(size int, from int) (ge.PublicationList, error) {
 
 	q := elastic.NewMatchAllQuery()
 
-	searchResult, err := client.Search().
+	service := client.Search().
 		Index("publications").
 		Query(q).
 		From(from).
-		Size(size).
-		//Pretty(true).
-		// Timeout("1000ms"). or
-		// Timeout(1000).
-		Do(ctx)
+		Size(size)
+
+	/*
+	// TODO: kind of kludged together here - probably much better way to do
+	agg := elastic.NewTermsAggregation().Field("type.label")
+	service = service.Aggregation("types", agg)
+
+	nested := elastic.NewNestedAggregation().Path("keywordList")
+	subAgg := nested.SubAggregation("keyword", elastic.NewTermsAggregation().Field("keywordList.label.keyword"))
+
+	nested2 := elastic.NewNestedAggregation().Path("affiliationList")
+	subAgg2 := nested2.SubAggregation("department",
+		elastic.NewTermsAggregation().Field("affiliationList.organization.label.dept"))
+
+	service = service.Aggregation("keywords", subAgg)
+	service = service.Aggregation("affiliations", subAgg2)
+    */
+
+	searchResult, err := service.Do(ctx)	
```

Comment:
@nymbyl  and I talked about these comments. these are for him to reference later and since this is an experiment, we are leaving it for now.